src/views/Package/index.tsx

```diff
@@ -1,63 +1,46 @@
 import { Box } from "@chakra-ui/react";
-import * as reflect from "jsii-reflect";
-import { useEffect, useState } from "react";
+import { useEffect } from "react";
 import { useParams, useLocation } from "react-router-dom";
 import { createAssembly } from "../../api/package/assemblies";
-import { parseSearch, getFullPackageName } from "../../api/package/util";
-import { GettingStarted } from "../../components/GettingStarted";
+import { fetchMetadata } from "../../api/package/metadata";
+import { parseSearch } from "../../api/package/util";
+import { PackageDetails } from "../../components/PackageDetails";
 import { PackageDocs } from "../../components/PackageDocs";
-import { PackageHeader } from "../../components/PackageHeader";
+import { useRequest } from "../../hooks/useRequest";
 
 interface PathParams {
   name: string;
   scope?: string;
   version: string;
 }
 
-interface PackageDocsHomeState {
-  assembly?: reflect.Assembly;
-  error?: string;
-}
-
 export function Package() {
   const { name, scope, version }: PathParams = useParams();
-  const [{ assembly }, setState] = useState<PackageDocsHomeState>({});
+  const [requestAssembly, assemblyResponse] = useRequest(createAssembly);
+  const [requestMetadata, metadataResponse] = useRequest(fetchMetadata);
   const q = parseSearch(useLocation().search);
 
   useEffect(() => {
-    createAssembly(name, version, scope)
-      .then((asm: reflect.Assembly) => {
-        setState({ assembly: asm });
-      })
-      .catch((err) => {
-        console.error(err);
-        setState((s) => ({ ...s, error: err }));
```

Comment:
error handling is built-in to `useRequest` and we can now access errors if needed via `assemblyResponse.error`